### PR TITLE
Many fixes of daterange class + add 33 unit tests for this

### PR DESCRIPTION
**Pull request by ddurieux**
This PR rewrite function resolve_daterange() in Timeperiod class to get the right information and to pass correct type and data arguments to dateRange classes.
Fix fonctions in daterange classes with the right type of arguments.
In end add many tests to test all cases of periods of function resolve_daterange()

### CODE REVIEW

File: alignak/daterange.py [hunk 1]
```diff
@@ -118,15 +105,8 @@ def find_day_by_offset(year, month, offset):
...
-
-    >>> find_day_by_offset(2015, "july", -1)
-    31
-
     """
-    month_id = Daterange.get_month_id(month)
-    if month_id is None:
-        return None
-    (tmp, days_in_month) = calendar.monthrange(year, month_id)
+    (tmp, days_in_month) = calendar.monthrange(year, month)
```
Comment (titilambert) [inline alignak/daterange.py]: (_, days_in_month) = calendar.monthrange(year, month)

File: alignak/daterange.py [hunk 2]
```diff
@@ -118,15 +105,8 @@ def find_day_by_offset(year, month, offset):
     :type offset: int
     :return: day number in the month
     :rtype: int
-
-    >>> find_day_by_offset(2015, "july", -1)
-    31
-
```
Comment (titilambert) [inline alignak/daterange.py]: Keep this please :)

File: alignak/daterange.py [hunk 3]
```diff
@@ -71,22 +71,9 @@ def find_day_by_weekday_offset(year, month, weekday, offset):
     :type offset: int
     :return: day number in the month
     :rtype: int
-
-    >>> find_day_by_weekday_offset(2010, "july", "tuesday", -1)
-    27
```
Comment (titilambert) [inline alignak/daterange.py]: Keep this please :)

**Comment by Seb-Solon** [comment]
Please add a short description of your rework. So that I can understand the case easily.

File: alignak/daterange.py [hunk 4]
```diff
@@ -72,21 +72,13 @@ def find_day_by_weekday_offset(year, month, weekday, offset):
     :return: day number in the month
     :rtype: int
 
+     :rtype: int
+
```
Comment (titilambert) [inline alignak/daterange.py]: I think you duplicate this line ...

File: alignak/daterange.py [hunk 5]
```diff
@@ -717,11 +712,11 @@ def is_correct(self):
         :rtype: bool
         """
         b = True
-        b &= self.swday in Daterange.weekdays
+        b &= self.swday in Daterange.weekdays.values()
```
Comment (Seb-Solon) [inline alignak/daterange.py]: Maybe it could be clearer with xrange(7) or 0<b<6.

File: alignak/objects/timeperiod.py [hunk 6]
```diff
@@ -804,49 +811,46 @@ def resolve_daterange(self, dateranges, entry):
...
-                                      0, emday, 0, 0, 0, other)
-                )
-                return
+            if not t1:
+                # print "Good catch 12"
+                if t0 in Daterange.weekdays:
+                    swday = Daterange.get_weekday_id(t0)
+                    swday_offset = t2
+                    ewday = swday
+                    ewday_offset = swday_offset
```
Comment (Seb-Solon) [inline alignak/objects/timeperiod.py]: Looks like swday_offset is not defined here. Should be t2 right? 
Strange this variale was used from the first regex :(

**Comment by Seb-Solon** [comment]
Looks good to me

**Pull request merged**